plugins/modules/fusion_volume.py

```diff
@@ -465,18 +378,11 @@ def validate_arguments(module, fusion):
             msg="'eradicate: true' cannot be used together with 'state: present'"
         )
 
-    if state == "present" and not volume:
-        # would create a volume; check size is lower than storage class limit
-        # (let resize checks be handled by the server since they are more complicated)
-        size = parse_number_with_metric_suffix(module, module.params["size"])
-        size_limit = get_storage_class(module, fusion).size_limit
-        if size > size_limit:
-            module.fail_json(
-                msg="Requested volume size {0} exceeds the storage class limit of {1}".format(
-                    module.params["size"],
-                    print_number_with_metric_suffix(size_limit),
-                )
-            )
+    size = parse_number_with_metric_suffix(module, module.params["size"])
```

Comment:
This gets run even if `module.params["state"] == "absent"`, no? In that case size is optional and parsing and checks probably shouldn't be done

Reply:
But for `destroyed=True` we have order as described on line 316->323, first we update all volume parameters and then only volume become destroyed. So this validation is required even for `state="absent"`

Reply:
Maybe put `if module.params["size"] is not None:` to make sure it doesn't fail if no `size` parameter is provided.

Reply:
@jankodera-ps But anyways, I am not agree with empty string, if we have in ansible task empty string in size field, then it is user's problem, if user didn't set size and it thrown an error, then it is a bug. The check should be done only for null value

Reply:
Yes, agreed, the contentious point was when size is not present at all